src/systemctl/systemctl.c

```diff
@@ -5661,6 +5668,31 @@ static int enable_unit(int argc, char *argv[], void *userdata) {
                 if (r < 0)
                         return r;
 
+                path = unit_dbus_path_from_name(*names);
+                if (!path)
+                        return log_oom();
+
+                r = sd_bus_call_method(
+                                bus,
+                                "org.freedesktop.systemd1",
+                                path,
+                                "org.freedesktop.DBus.Properties",
+                                "GetAll",
+                                &error,
+                                &reply,
+                                "s", "");
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get properties: %s", bus_error_message(&error, r));
+
+                r = bus_message_map_all_properties(reply, property_map, &info);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to map properties: %s", bus_error_message(&error, r));
+
+                if (STR_IN_SET(verb, "mask", "unmask") && streq_ptr(info.load_state, "not-found") && streq_ptr(info.active_state, "inactive")) {
```

Comment:
I think we should only invoke the unit_exists() stuff when the verb is actually mask or unmask...